### PR TITLE
Adds event API to basic-details form

### LINKED ISSUE #462: Implement event creation using API
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

Implement event creation using the new API.

Reference: https://open-event-api.herokuapp.com/#events-events-collection-post

### LINKED ISSUE #554: Implement basic event api on basic-details form
Parent Issue #462

### PR DESCRIPTION
**Pull request by geekyd**
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

[Work In Progress]
#### Changes proposed in this pull request:

- Adds event creation API to basic-details form

Fixes #554

### CODE REVIEW

**Comment by geekyd** [comment]
@niranjan94 Everything seems to work fine for the basic-details form, except the tickets & social links. Please guide. Thanks

File: app/serializers/event.js [hunk 1]
```diff
@@ -9,5 +10,14 @@ export default ApplicationSerializer.extend(CustomPrimaryKeyMixin, {
     topic     : 'event-topic',
     subTopic  : 'event-sub-topic',
     copyright : 'event-copyright'
+  },
+  serialize(snapshot, options) {
```
Comment (niranjan94) [inline app/serializers/event.js]: Why are we omitting these? 
And if you want to ignore some attributes, you use readonly option in the model. 

hello: attr('string', { readonly: true}) ;

I think I added it

**Comment by niranjan94** [comment]
- First save the event model. 
- And after that event model is saved, save the social link and ticket.

**Comment by niranjan94** [comment]
@geekyd any progress on this ?

**Comment by geekyd** [comment]
@niranjan94 I'm working on this, but the new endpoints are giving error `403` eg `v1/tickets`, the old endpoints are still working, also the event-topic, type, subtopics are no longer getting saved. Please guide. Should we support the old endpoints?

Please guide.

**Comment by niranjan94** [comment]
What exactly is the 403 error for the new endpoint? 

No. We will only use old endpoints for GET and nothing else. 

If subtopic, topic, type etc is not saving open an issue on the server

**Comment by geekyd** [comment]
@niranjan94 Here is the response that I get when I try to add ticket

Endpoint
```
/v1/tickets
```

Payload
```
{
   "data":{
      "attributes":{
         "name":"FREE",
         "type":"free",
         "price":null,
         "quantity":100,
         "description":null,
         "is-description-visible":true,
         "is-hidden":false,
         "sales-starts-at":"2017-07-24T18:30:00.000Z",
         "sales-ends-at":"2017-08-03T18:30:00.000Z",
         "min-order":1,
         "max-order":10,
         "is-fee-absorbed":true,
         "position":null
      },
      "relationships":{
         "event":{
            "data":{
               "type":"event",
               "id":"167"
            }
         }
      },
      "type":"ticket"
   }
}
```
Response
```
{
   "jsonapi":{
      "version":"1.0"
   },
   "errors":[
      {
         "status":403,
         "detail":"Access forbidden",
         "source":{
            "source":""
         },
         "title":"Access Forbidden"
      }
   ]
}
```

**Comment by niranjan94** [comment]
@geekyd open an issue for this too.

**Comment by geekyd** [comment]
@niranjan94 Opened https://github.com/fossasia/open-event-orga-server/issues/4157

**Comment by niranjan94** [comment]
@geekyd lets tackle these two later. For now, let the basic event details be saved and once that is done lets merge this PR. We'll not wait for the server anymore. When server is fixed, we'll open a new issue and do it.

**Comment by geekyd** [comment]
@niranjan94 Okay. I would suggest instead of closing the issue we should create child issues for #462 & work on it as the server gets ready. What do you suggest?

**Comment by geekyd** [comment]
@niranjan94 Please review. Thanks

File: app/routes/create.js [hunk 2]
```diff
@@ -29,10 +29,38 @@ export default Route.extend(AuthenticatedRouteMixin, EventWizardMixin, {
   },
   actions: {
     save() {
-
+      this.get('currentModel.data.event').save()
+        .then(data => {
+          let savedTicketsPromises = this.get('currentModel.data.event.tickets').toArray().map(ticket => ticket.save());
+          let savedSocialPromises = this.get('currentModel.data.event.socialLinks').toArray().map(link => link.save());
+          let allPromises = [savedTicketsPromises, savedSocialPromises];
```
Comment (niranjan94) [inline app/routes/create.js]: This variable is not needed. Directly pass the array.

**Pull request merged**